lib/resty/core/shdict.lua

```diff
@@ -380,9 +380,31 @@ local function shdict_incr(zone, key, value, init)
         init = 0
     end
 
+    if exptime ~= nil then
+        local typ = type(exptime)
+        if typ ~= "number" then
+            exptime = tonumber(exptime)
+
+            if not exptime then
+                error("bad exptime arg: number expected, got " .. typ, 2)
+            end
+        end
+
+        if exptime < 0 then
+            error('bad "exptime" argument', 2)
+        end
+
+        if has_init == 0 then
+            error('must provide "init" when providing "exptime"', 2)
```

Comment:
This is a new case of an shdict API both creating values and updating already existing ones, and now having to deal with conditionally setting their exptime as well. 
Unlike `set()` and `expire()` which unconditionally replace the exptime, I made `incr()` only replace it when the value did not exist. If you believe this does not make sense, let me know.